
This patch adds Quake packages to buildroot:
* Quake engine itself (SDL port)
* a shareware data file (PAK/WAD) to test the engine

It is inspired of OpenEmbedded work.

sdlquake.mk could be simplier if a target FOO_HOOK_POST_INSTALL was
existing in Makefile.autotools.in.

Still searching a way to apply sdlquake-1.0.9-no-x86-asm.patch on all arch
EXCEPT x86.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/package/games/sdlquake/sdlquake.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/games/sdlquake/sdlquake.mk	2011-08-08 12:14:46.000000000 +0200
@@ -0,0 +1,36 @@
+#############################################################
+#
+# Quake 1 (SDL_Port)
+#
+#############################################################
+
+SDLQUAKE_VERSION = 1.0.9
+SDLQUAKE_SOURCE = sdlquake-$(SDLQUAKE_VERSION).tar.gz
+SDLQUAKE_SITE = http://www.libsdl.org/projects/quake/src/
+#SDLQUAKE_AUTORECONF = NO
+SDLQUAKE_INSTALL_STAGING = NO
+SDLQUAKE_INSTALL_TARGET = YES
+
+SDLQUAKE_CONF_OPT = \
+	--target=$(GNU_TARGET_NAME) \
+	--host=$(GNU_TARGET_NAME) \
+	--build=$(GNU_HOST_NAME) \
+	--with-sdl-prefix=$(STAGING_DIR)/usr \
+	--prefix=/usr \
+	--sysconfdir=/etc
+
+SDLQUAKE_DEPENDENCIES = sdl sdl_mixer SDL_net
+
+$(eval $(call AUTOTARGETS,package/games,sdlquake))
+
+NAMEVER = $($(PKG)_NAME)-$($(PKG)_VERSION)
+$(SDLQUAKE_TARGET_PATCH):
+	$(call MESSAGE,"Patching $($(PKG)_DIR_PREFIX)/$($(PKG)_NAME)")
+	toolchain/patch-kernel.sh $($(PKG)_DIR) $($(PKG)_DIR_PREFIX)/$($(PKG)_NAME) $(NAMEVER)\*.patch $(NAMEVER)\*.patch.$(ARCH) || exit 1;
+	(cd $(SDLQUAKE_DIR); \
+	touch INSTALL NEWS README AUTHORS ChangeLog; \
+	echo "AC_PROG_CC" >> configure.in; \
+	aclocal; \
+	autoconf; \
+	automake --add-missing;)
+	touch $@
Index: buildroot/package/games/sdlquake/sdlquake-1.0.9-no-x86-asm.patch.arm
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/games/sdlquake/sdlquake-1.0.9-no-x86-asm.patch.arm	2011-08-08 12:14:46.000000000 +0200
@@ -0,0 +1,35 @@
+--- /tmp/Makefile.am	2008-08-11 20:31:09.000000000 +0200
++++ sdlquake-1.0.9/Makefile.am	2008-08-11 20:55:00.263198000 +0200
+@@ -33,7 +33,6 @@
+ 	crc.h			\
+ 	cvar.c			\
+ 	cvar.h			\
+-	d_copy.S		\
+ 	d_edge.c		\
+ 	d_fill.c		\
+ 	d_iface.h		\
+@@ -48,7 +47,6 @@
+ 	d_sprite.c		\
+ 	d_surf.c		\
+ 	d_zpoint.c		\
+-	dosasm.S		\
+ 	dosisms.h		\
+ 	draw.c			\
+ 	draw.h			\
+@@ -103,7 +101,6 @@
+ 	r_sprite.c		\
+ 	r_surf.c		\
+ 	r_vars.c		\
+-	r_varsa.S		\
+ 	render.h		\
+ 	resource.h		\
+ 	sbar.c			\
+@@ -136,7 +133,7 @@
+ 	world.h			\
+ 	zone.c			\
+ 	zone.h			\
+-	$(X86_SRCS) $(NONX86_SRCS)
++	$(NONX86_SRCS)
+ 
+ X86_SRCS =			\
+ 	snd_mixa.S		\
Index: buildroot/package/games/sdlquake/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/games/sdlquake/Config.in	2011-08-08 12:14:46.000000000 +0200
@@ -0,0 +1,10 @@
+config BR2_PACKAGE_SDLQUAKE
+	bool "sdlquake"
+	select BR2_PACKAGE_SDL
+	select BR2_PACKAGE_SDL_MIXER
+	select BR2_PACKAGE_SDL_NET
+	help
+	  sdlquake is a port of ID Software's popular Quake 3-D shooter game.
+	  (Not supported by ID Software)
+
+	  http://www.libsdl.org/projects/quake/
Index: buildroot/package/games/quake-pak/quake-pak.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/games/quake-pak/quake-pak.mk	2011-08-08 12:14:46.000000000 +0200
@@ -0,0 +1,34 @@
+#############################################################
+#
+# Quake shareware data files (PAK/WAD)
+#
+#############################################################
+
+QUAKE_PAK_VERSION = 1.0.6
+QUAKE_PAK_SOURCE = quakesw-$(QUAKE_PAK_VERSION).tar.gz
+QUAKE_PAK_SITE = http://www.libsdl.org/projects/quake/data/
+QUAKE_PAK_CAT = $(ZCAT)
+QUAKE_PAK_TARGET_DIR = $(TARGET_DIR)/usr/share/games/quake/
+
+$(DL_DIR)/$(QUAKE_PAK_SOURCE):
+	 $(WGET) -P $(DL_DIR) $(QUAKE_PAK_SITE)/$(QUAKE_PAK_SOURCE)
+
+quake-pak-source: $(DL_DIR)/$(QUAKE_PAK_SOURCE)
+
+$(QUAKE_PAK_TARGET_DIR)/id1/pak0.pak: $(DL_DIR)/$(QUAKE_PAK_SOURCE)
+	mkdir -p $(QUAKE_PAK_TARGET_DIR)
+	$(QUAKE_PAK_CAT) $(DL_DIR)/$(QUAKE_PAK_SOURCE) | tar -C $(QUAKE_PAK_TARGET_DIR) $(TAR_OPTIONS) -
+
+quake-pak: $(QUAKE_PAK_TARGET_DIR)/id1/pak0.pak
+
+quake-pak-dirclean:
+	rm -rf $(QUAKE_PAK_TARGET_DIR)/id1/
+
+#############################################################
+#
+# Toplevel Makefile options
+#
+#############################################################
+ifeq ($(strip $(BR2_PACKAGE_QUAKE_PAK)),y)
+TARGETS+=quake-pak
+endif
Index: buildroot/package/games/quake-pak/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/games/quake-pak/Config.in	2011-08-08 12:14:46.000000000 +0200
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_QUAKE_PAK
+	bool "shareware Quake PAK/WAD file"
+	depends on BR2_PACKAGE_SDLQUAKE
+	help
+	  This will install the shareware pak/wad data file for the Quake game.
+
+	  The file will be placed in the /usr/share/games/quake/ directory.
Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2011-08-08 12:14:41.000000000 +0200
+++ buildroot/package/Config.in	2011-08-08 12:14:46.000000000 +0200
@@ -87,8 +87,10 @@
 menu "Games"
 source "package/games/gnuchess/Config.in"
 source "package/games/prboom/Config.in"
+source "package/games/quake-pak/Config.in"
 source "package/games/doom-wad/Config.in"
 source "package/games/rubix/Config.in"
+source "package/games/sdlquake/Config.in"
 endmenu
 
 menu "Graphic libraries and applications (graphic/text)"
